projects/packages/search/src/dashboard/components/record-meter/notice-box.jsx

```diff
@@ -1,94 +1,155 @@
-	if (
-		props.recordCount > props.planRecordLimit * CLOSE_TO_LIMIT_PERCENT &&
-		props.recordCount < props.planRecordLimit
-	) {
-		notices.push( {
+			link: {
+				text: __( 'learn more', 'jetpack-search-pkg' ),
+				url: 'https://jetpack.com/support/search/product-pricing/',
+			},
+		},
+		5: {
+			id: 5,
 			message: sprintf(
 				// translators: %d: site's current plan record limit
 				__(
 					"You're close to the max amount of records for this billing tier. Once you hit %d indexed records, you'll automatically be billed for the next tier",
 					'jetpack-search-pkg'
 				),
-				props.planRecordLimit
+				planRecordLimit
 			),
-		} );
-	}
+			link: {
+				text: __( 'learn more', 'jetpack-search-pkg' ),
+				url: 'https://jetpack.com/support/search/product-pricing/',
+			},
+		},
+	};
+};
 
-	if ( ! notices || notices.length < 1 ) {
+/**
+ * Returns a notice box for displaying notices about record count and plan limits
+ *
+ * @param {object} props - Props
+ * @param {number} props.recordCount - Current count of user's total records
+ * @param {number} props.recordLimit - Max number of records allowed in user's current tier
+ * @param {boolean} props.hasBeenIndexed - True if site has a last indexed date
+ * @param {boolean} props.hasValidData - True if data is present and in valid form
+ * @param {boolean} props.hasItems - True if there is at least one indexed record
+ * @returns {React.Component} notice box component.
+ */
+export function NoticeBox( props ) {
+	const activeNoticeIds = [];
+	const NOTICES = getNotices( props.planRecordLimit );
+	const [ showNotice, setShowNotice ] = useState( true );
+
+	// deal with localStorage for ensuring dismissed notice boxs are not re-displayed
+	const dismissedNoticesString = localStorage.getItem( DISMISSED_NOTICES ) ?? '';
+
+	const DATA_NOT_VALID = '1',
+		HAS_NOT_BEEN_INDEXED = '2',
+		NO_INDEXABLE_ITEMS = '3',
+		OVER_RECORD_LIMIT = '4',
+		CLOSE_TO_LIMIT = '5';
+
+	// check if data is valid
+	props.hasValidData === false &&
+		! dismissedNoticesString.includes( DATA_NOT_VALID ) &&
+		activeNoticeIds.push( DATA_NOT_VALID );
+
+	// check site has been indexed
+	props.hasBeenIndexed === false &&
+		! dismissedNoticesString.includes( HAS_NOT_BEEN_INDEXED ) &&
+		activeNoticeIds.push( HAS_NOT_BEEN_INDEXED );
+
+	// check at least one indexable item
+	props.hasItems === false &&
+		! dismissedNoticesString.includes( NO_INDEXABLE_ITEMS ) &&
+		activeNoticeIds.push( NO_INDEXABLE_ITEMS );
+
+	// check if over limit
+	props.recordCount > props.planRecordLimit &&
+		! dismissedNoticesString.includes( OVER_RECORD_LIMIT ) &&
+		activeNoticeIds.push( OVER_RECORD_LIMIT );
+
+	// check if close to reaching limit
+	props.recordCount > props.planRecordLimit * CLOSE_TO_LIMIT_PERCENT &&
+		props.recordCount < props.planRecordLimit &&
+		! dismissedNoticesString.includes( CLOSE_TO_LIMIT ) &&
+		activeNoticeIds.push( CLOSE_TO_LIMIT );
+
+	if ( ! activeNoticeIds || activeNoticeIds.length < 1 || ! showNotice ) {
```

Comment:
```suggestion
	if ( activeNoticeIds.length < 1 || ! showNotice ) {
```

We're instantiating `activeNoticeIds` in line 83, so we have no need to check if it'll be truthy or not 🙂